hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -2107,7 +2107,6 @@ public boolean contains(Object object) {
 
 	@Override
 	public boolean contains(String entityName, Object object) {
-		checkOpen();
```

Comment:
what about changing to checkOpenOrWaitingForAutoClose()?

Reply:
@dreab8 Good point. It works with `checkOpenOrWaitingForAutoClose`.

Reply:
@dreab8 we don't need it to be open in the proxy case?

Reply:
We also need to check more use cases since `checkOpenOrWaitingForAutoClose` is tied to JPA bootstrap only, which happens to be covered by the current unit tests. I'll add some tests which bootstrap Hibernate natively so we can make sure we cover that use case as well.

Reply:
Well, we can't test it for native bootstrap since the Session is closed anyway and the undelrying Connetcion as well.

So, the current solution uses `checkOpenOrWaitingForAutoClose` like many other Sesison methods. If anyone agrees, we can integrate it as is since all tests pass just fine.

Reply:
In case of Session it have to fail so if this happens I think it should work.

Reply:
I don't understand what exactly you ate referring to. Everything happen inside the SessionImpl class. Are you saying that it's fine to fail for Hibernate native bootstrap?

Reply:
Yes it is what I mean.

Reply:
Anyway I want to investigate a little more the issue, so please wait before merging.

Reply:
Sure thing.

Reply:
@vladmihalcea It seems good to me.

Reply:
@dreab8 Cool. Then I'll integrate it into master and 5.2.